src/par_iter/test.rs

```diff
@@ -272,6 +272,144 @@ pub fn check_range_exact_and_bounded() {
 }
 
 #[test]
+pub fn check_cmp() {
+    let a: Vec<usize> = (0..1024).collect();
+    let b: Vec<usize> = (0..1024).collect();
+
+    let result = a.cmp(&b);
+
+    assert!(result == ::std::cmp::Ordering::Equal);
+}
+
+#[test]
+pub fn check_cmp_lt() {
+    let a: Vec<usize> = (0..1024).collect();
+    let b: Vec<usize> = (1..1024).collect();
+
+    let result = a.cmp(&b);
+
+    assert!(result == ::std::cmp::Ordering::Less);
+}
+
+#[test]
+pub fn check_cmp_gt() {
+    let a: Vec<usize> = (1..1024).collect();
+    let b: Vec<usize> = (0..1024).collect();
+
+    let result = a.cmp(&b);
+
+    assert!(result == ::std::cmp::Ordering::Greater);
+}
+
+#[test]
+pub fn check_partial_cmp() {
+    let a: Vec<usize> = (0..1024).collect();
+    let b: Vec<usize> = (0..1024).collect();
+
+    let result = a.partial_cmp(&b);
+
+    assert!(result == Some(::std::cmp::Ordering::Equal));
+}
+
+#[test]
+pub fn check_partial_cmp_lt() {
+    let a: Vec<usize> = (0..1024).collect();
+    let b: Vec<usize> = (1..1024).collect();
+
+    let result = a.partial_cmp(&b);
+
+    assert!(result == Some(::std::cmp::Ordering::Less));
+}
+
+#[test]
+pub fn check_partial_cmp_gt() {
+    let a: Vec<usize> = (1..1024).collect();
+    let b: Vec<usize> = (0..1024).collect();
+
+    let result = a.partial_cmp(&b);
+
+    assert!(result == Some(::std::cmp::Ordering::Greater));
+}
+
+#[test]
+pub fn check_partial_cmp_none() {
+    let mut a: Vec<f32> = (0..1024u16).map(From::from).collect();
+    let b: Vec<f32> = (0..1024u16).map(From::from).collect();
+
+    a[0] = ::std::f32::NAN;
+
+    let result = a.partial_cmp(&b);
+
+    assert!(result == None);
+}
+
+#[test]
+pub fn check_eq() {
+    let a: Vec<usize> = (0..1024).collect();
```

Comment:
I'll add tests both between pars and between par and seq iterators. That way you'll have tests for expected behavior internally as well as expected behavior as compared to the external standard of seq iterators.